### PR TITLE
Upgrade libp2p to 0.46.1

### PR DESCRIPTION
**Pull request by kpp**
Upgrade libp2p to 0.46.1 which reduces the number of libp2p-core crates. Right now we compile both v0.32.0 and v0.33.0. This PR fixes it.

Related issue: https://github.com/libp2p/rust-libp2p/issues/2715

### CODE REVIEW

**Review by tomaka — APPROVED** [review]

**Review by andresilva — APPROVED** [review]

**Comment by andresilva** [comment]
@tomaka I think your review didn't count for the check reviews CI task because you are not part of the core devs team. We'll need another approval.

**Review by bkchr — APPROVED** [review]

**Comment by andresilva** [comment]
bot merge

**Comment by xlc** [comment]
Can we backport this to 0.9.26?

**Pull request merged**
